src/MCPClient/lib/clientScripts/extract_contents.py

```diff
@@ -167,6 +167,8 @@ def main(job, transfer_uuid, sip_directory, date, task_uuid, delete=False):
         # the names we want to preserve in our PREMIS:originalLocation.
         temp_dir = temporary_directory(output_file_path, date)
         rename(output_file_path, temp_dir)
+        file_.currentlocation = temporary_directory(file_.currentlocation, date)
```

Comment:
I wonder if we could extend `fileOperations.rename(...)` with an option to update the current location? It seems like the two could be linked in such a way, but I'm not sure where else in the code we currently perform something similar to this?

Reply:
Yes, maybe this is a good idea. I have searched through the code and found the following:

- Places where `fileOperations.py::rename` is used:
  - extract_contents.py
    - `fileOperations.py::updateFileLocation` is NOW also used (post
      ce4e41b0f7811d0d96997703537878d13ee16076)
  - move_dspace_license_files_to_dspace_licenses.py
    - `fileOperations.py::updateFileLocation` IS also used
  - move_dspace_mets_files_to_dspace_mets.py
    - `fileOperations.py::updateFileLocation` IS also used
  - check_for_access_directory.py
    - `fileOperations.py::updateFileLocation` IS also used
  - move_transfer.py
    - `fileOperations.py::updateFileLocation` is NOT also used, BUT it is
      moving an entire transfer
  - move_sip.py
    - `fileOperations.py::updateFileLocation` is NOT also used, BUT it is
      moving an entire SIP

So if we were to call `fileOperations.py::updateFileLocation` from within `fileOperations.py::rename` then we would have to:

1. prevent the calling of `updateFileLocation` in the transfer/SIP cases; and
2. make sure that `updateFileLocation` is being used in the same way in the other four cases and add parameterization where it is not.

I'm not entirely sure it is worth the effort, but if you think it is I could do it.

This comment did, however, lead me to add a call to `updateFileLocation` in `extract_contents.py` which was needed: now the package files should have PREMIS events of type `'movement'`, which was definitely needed I think:

    <premis:eventType>movement</premis:eventType>
    <premis:eventDateTime>2018-08-28T22:39:00.360525+00:00</premis:eventDateTime>
    <premis:eventDetail/>
    <premis:eventOutcomeInformation>
      <premis:eventOutcome/>
      <premis:eventOutcomeDetail>
        <premis:eventOutcomeDetailNote>
          moved from="%transferDirectory%objects/one_file.zip"; moved to="%transferDirectory%objects/one_file.zip-AM-DATE-SFX-2018-08-28T22:39:00.168851+00:00"
        </premis:eventOutcomeDetailNote>
      </premis:eventOutcomeDetail>
    </premis:eventOutcomeInformation>

One thing I noticed was that the `'unpacking'` events of the unpacked files now reference the unmodified names of the packages that they were unpacked from. See below. I think this is ok, but let me know if you think otherwise.

    <premis:eventType>unpacking</premis:eventType>
    <premis:eventDateTime>2018-08-28T22:39:00.337993+00:00</premis:eventDateTime>
    <premis:eventDetail>
      Unpacked from: %transferDirectory%objects/one_file.zip (ac88a562-e26a-4b2b-b31b-02b27289515e)
    </premis:eventDetail>